File: Java/hellointerview/01-two-pointers/moveZeroes.java
public class Solution {
  public void moveZeroes(int[] nums) {
    int currentNonZeroSpot = 0;
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] != 0) {
        var temp = nums[i];
        nums[i] = nums[currentNonZeroSpot];
        nums[currentNonZeroSpot] = temp;
        currentNonZeroSpot++;
      }
    }
  }
}

File: Go/hellointerview/01-two-pointers/moveZeroes.go
package main

func MoveZeroes(nums []int) {
	nextNonZeroSpot := 0

	for i, n := range nums {
		if n != 0 {
			nums[nextNonZeroSpot], nums[i] = nums[i], nums[nextNonZeroSpot]
			nextNonZeroSpot++
		}
	}
}
